modules/ROOT/pages/mule-data-types.adoc

```diff
@@ -0,0 +1,33 @@
+== Mule Data Types
+
+Mule runtime engine uses a typed data structure called `TypedValue` that centralizes data access across different components. This data structure provides access to both the actual data and its data type. The data type consists of two values, the MIME type and the encoding of the data.
+
+=== Input Data
+
+Mule stores all variables and the payload using the `TypedValue` data structure. Also, operations like `List` from the File connector return their result as a `TypedValue` instance, enabling DataWeave to list different types of files (JSON, XML, CSV, and so on) to read.
+
+=== Output Data
+
+DataWeave always outputs a `TypedValue` instance. DataWeave decides what writer to use based on the data type of the `TypedValue` instance. However, you can manually specify the data type in a DataWeave script by using the `output` directive.
+If the script that Mule is executing is assigned to a message processing field, Mule informs  DataWeave what is the expected type based on the metadata of that field. For example, if it is a Plain Old Java Object (Pojo) then DataWeave knows what class to instantiate and to use the Java Writer to generate the output.
```

Comment:
```suggestion
If the script that Mule is executing is assigned to a message processing field, Mule informs DataWeave which type of data to expect type based on the metadata of that field. For example, if <<the metadata? the data type?>> is a Plain Old Java Object (Pojo), then DataWeave knows not only which class to instantiate but also that it must use the Java writer to generate the output.
```

Reply:
Whether to use `what` or `which` is one of those tricky questions in English grammar. Basically, If you are trying to make a choice, `what` is used to ask when there are _an unknown number or infinite possibilities_ for an answer. `Which` is used if you are choosing between a _more limited number of items, already defined_.